### PR TITLE
Handle value overflow checks for IonParser.getIntValue() - #428

### PR DESCRIPTION
**Pull request by thomasdelange5**
Add number size checks for IonParser getIntValue and getLongValue. Fix failing tests for #428

### CODE REVIEW

File: ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java
Comment (cowtowncoder) [inline ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java:410]: Let's not add this back (I think @tgregg is working on handling these edge cases)

**Comment by cowtowncoder** [comment]
Looks good @thomasdelange5 -- thank you for contributing this!

Only one other thing before I can merge this: if we haven't gotten CLA for you yet (it only needs to be done once per contributor):

https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf

(or possibly alternate Corporate CLA addendum if this is under some company's; we have a few Amazon contributors for Ion).
It is usually easiest to fill & sign, print, scan/photo, email to `cla` at fasterxml dot com.
Once we have it (and if you have sent previously, let me know please), I can proceed with merging this fix!

File: ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java
Comment (tgregg) [reply ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java:410]: Correct. If fuzzing identifies any new leaked exceptions I'd like to fix those in ion-java; in the meantime we can cover them in `failing` tests.

**Comment by thomasdelange5** [comment]
Thank you for getting back so quickly @cowtowncoder! I will remove that catch clause and submit the CLA now.

File: ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java
Comment (tgregg) [inline ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java:393]: Minor: for performance reasons, I recommend storing this value and using it in both branches. ion-java has to do some bounds checking on the values, and we should avoid doing that twice when possible.

Comment (thomasdelange5) [reply ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java:393]: Ok will do. Thanks @tgregg

Comment (cowtowncoder) [reply ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java:393]: was about to suggest same, +1

**Review by tgregg — APPROVED** [review]

**Review by cowtowncoder — APPROVED** [review]

**Comment by cowtowncoder** [comment]
CLA received, will merge for 2.17(.0).

Thank you, @thomasdelange5 !

**Comment by thomasdelange5** [comment]
Awesome, thank you very much!

**Pull request merged**
